### ISSUE TITLE
Code Insights: Conduct design UI review of compute powered insight creation UI

### ISSUE THREAD

**Issue by vovakulikov** [sourcegraph/sourcegraph#38836]
Follow up for https://github.com/sourcegraph/sourcegraph/pull/38762

## Test plan
- Go to the /insights/create/group-results URL.
- Make sure that you can create a compute-powered insight through the newly created compute creation UI page and see a correct live preview result.

**Comment by AlicjaSuska** [comment]
@vovakulikov  thank you for submitting this issue. Here is the design review
- the container of the whole form is wider than for other code insight types - it should be narrower
- we should add 'type: X | change type' element to all remaining insight categories for consistency
- change the helper text in the header, remove the helper text below the cards ([design](https://www.figma.com/file/DqZHetdfqHSVzzWtyfjOmr/Compute-x-Insights%2C-RFC-696%2C-%2337923-%5BReady-fro-dev%5D?node-id=1933%3A21380))
- move the extensions type next to the language type ([design](https://www.figma.com/file/DqZHetdfqHSVzzWtyfjOmr/Compute-x-Insights%2C-RFC-696%2C-%2337923-%5BReady-fro-dev%5D?node-id=1933%3A21380))
- author and commit should be disabled when the form is empty and become available only when the user adds diff or commit search. Now, it's enabled when the form is empty, then becomes disabled when the user adds a non-commit non-diff search.
- map result buttons shouldn't all have rounded corners. It's a segmented buttons, radius of majority of corners should be 0. Radii are correct when all the buttons are active.
<img width="299" alt="Screenshot 2022-07-19 at 14 45 54" src="https://user-images.githubusercontent.com/20326070/179753631-5d2294e6-5df1-42e4-bb57-bdc4762d3afa.png">

- when chart has many series with long names, everything becomes displaced. We should at least wrap the series in the key to avoid horizontal scroll in this creation flow. @vovakulikov we can work on the improvements to the chart itself (better placement of data series names) in a separate issue.
<img width="601" alt="Screenshot 2022-07-19 at 14 47 31" src="https://user-images.githubusercontent.com/20326070/179754150-3f673475-ae17-4103-875a-d2cfe5ba6fe9.png">

- general flow-related thought: we load the live preview only after user fills out both query and the name. Maybe we could load it after entering only the search query and use a temporary name 'data series 1' until user enters their own name? Not necessary for this issue, but I think it would make the flow smoother.

- let's reduce the space between the header and 'Targeted repositories' to 40px
<img width="983" alt="Screenshot 2022-07-19 at 14 54 20" src="https://user-images.githubusercontent.com/20326070/179755375-41faf134-d4dc-4b7d-bfa1-f5b691660f11.png">

- the 'clear all fields' button changed color in this version (should be blue text)
<img width="986" alt="Screenshot 2022-07-19 at 14 54 04" src="https://user-images.githubusercontent.com/20326070/179755434-a9d50cec-eb72-4791-a7e1-d920c8637d2d.png">


Important: I couldn't test how compute charts look and behave on the dashboard - @vovakulikov I'm getting the same error as we were getting yesterday:
<img width="949" alt="Screenshot 2022-07-19 at 15 00 38" src="https://user-images.githubusercontent.com/20326070/179756339-180065b9-ece7-4d3c-a060-9b8edbf32c0d.png">

**Comment by vovakulikov** [comment]
Thanks for the review @AlicjaSuska 

- These three will be addressed in this issue. https://github.com/sourcegraph/sourcegraph/issues/37218 I'll add them to the issue's description 

> - the container of the whole form is wider than for other code insight types - it should be narrower
 > - we should add 'type: X | change type' element to all remaining insight categories for consistency
> - change the helper text in the header, remove the helper text below the cards ([design](https://www.figma.com/file/DqZHetdfqHSVzzWtyfjOmr/Compute-x-Insights%2C-RFC-696%2C-%2337923-%5BReady-fro-dev%5D?node-id=1933%3A21380))

- We missed this while was adding a new card for this page. I created a follow-up for this. https://github.com/sourcegraph/sourcegraph/issues/39064

> move the extensions type next to the language type ([design](https://www.figma.com/file/DqZHetdfqHSVzzWtyfjOmr/Compute-x-Insights%2C-RFC-696%2C-%2337923-%5BReady-fro-dev%5D?node-id=1933%3A21380))

- This one has been addressed already in upcoming PR here https://github.com/sourcegraph/sourcegraph/pull/39021

> author and commit should be disabled when the form is empty and become available only when the user adds diff or commit search. Now, it's enabled when the form is empty, and then becomes disabled when the user adds a non-commit non-diff search.

- Unfortunately this is a bug inside of the wildcard button groups component. I will reach out @sourcegraph/frontend-devs teams about it.

>map result buttons shouldn't all have rounded corners. It's a segmented buttons, radius of majority of corners should be 0. Radii are correct when all the buttons are active.

- I 100% agree about this. Legend items layout has been fixed in this PR. https://github.com/sourcegraph/sourcegraph/pull/39021. The label layout problem will be addressed in this issue https://github.com/sourcegraph/sourcegraph/issues/39032

>when chart has many series with long names, everything becomes displaced. We should at least wrap the series in the key to avoid horizontal scroll in this creation flow. @vovakulikov we can work on the improvements to the chart itself (better placement of data series names) in a separate issue.

- Interesting thought. I agree with it, and I believe we even had a conversation about it before. I filled up an issue for it here https://github.com/sourcegraph/sourcegraph/issues/39066

> general flow-related thought: we load the live preview only after user fills out both query and the name. Maybe we could load it after entering only the search query and use a temporary name 'data series 1' until user enters their own name? Not necessary for this issue, but I think it would make the flow smoother.

- Also will be done in redesign issue since we don't have time to include it in this version
> let's reduce the space between the header and 'Targeted repositories' to 40px
> the 'clear all fields' button changed color in this version (should be blue text)

- As we discussed in DM, this issue appeared only on the local machines, and we couldn't catch this problem on k8s. Since this issue was fixed yesterday, I suspect something went wrong with the local git setup. 

>Important: I couldn't test how compute charts look and behave on the dashboard - @vovakulikov I'm getting the same error as we were getting yesterday:

**Comment by AlicjaSuska** [comment]
Review for the insights on dashboards:
- labels below the chart are overlapping (@vovakulikov I assume it will be a separate issue)
- everything works as expected for this version
- one small UI issue: the gradient on the key appears only after a scroll. It should be visible from the beginning ([loom](https://www.loom.com/share/16d01095d7ac4a8fa4458639ba6129f4))

**Comment by vovakulikov** [comment]
Thanks, @AlicjaSuska, for the second round of UI review. 

I recently opened an issue for it here https://github.com/sourcegraph/sourcegraph/issues/39032
>labels below the chart are overlapping (@vovakulikov I assume it will be a separate issue)

Oh, thanks. I opened an issue for it as well here https://github.com/sourcegraph/sourcegraph/issues/39139
>one small UI issue: the gradient on the key appears only after a scroll. It should be visible from the beginning ([loom](https://www.loom.com/share/16d01095d7ac4a8fa4458639ba6129f4))

**Comment by vovakulikov** [comment]
@AlicjaSuska, I think we can close this issue as we discussed in DM. Thanks once again for the comprehensive and clear UI review for the compute insight. I'm glad we capture and will track all problems from the first version and fix them in the next iteration over compute-powered insight.
